lib/SILOptimizer/Transforms/CopyPropagation.cpp

```diff
@@ -681,36 +44,33 @@ class CopyPropagation : public SILFunctionTransform {
 
 /// Top-level pass driver.
 void CopyPropagation::run() {
-  LLVM_DEBUG(llvm::dbgs() << "*** CopyPropagation: " << getFunction()->getName()
-                          << "\n");
+  auto *f = getFunction();
+  
+  // Debug label for unit testing.
+  LLVM_DEBUG(llvm::dbgs() << "*** CopyPropagation: " << f->getName() << "\n");
 
   // This algorithm fundamentally assumes ownership.
-  if (!getFunction()->hasOwnership())
+  if (!f->hasOwnership())
     return;
 
-  // Step 1. Find all copied defs.
-  CopyPropagationState pass(getFunction());
-  SmallSetVector<SILValue, 16> copiedDefs;
-  for (auto &bb : *pass.func) {
+  // Driver: Find all copied defs.
+  llvm::SmallSetVector<SILValue, 16> copiedDefs;
+  for (auto &bb : *f) {
     for (auto &i : bb) {
       if (auto *copy = dyn_cast<CopyValueInst>(&i))
-        copiedDefs.insert(stripCopies(copy));
+        copiedDefs.insert(getCanonicalCopiedDef(copy));
     }
   }
-
+  // Perform copy propgation for each copied value.
+  CanonicalOSSALifetime lifetime;
```

Comment:
For instance (noting my comment about CanonicalOSSALifetime is confusing), I think this code would be clearer if it was this:

```
OSSALifetimeCanonicalizer canonicalizer;
```

or

```
OSSALifetimeCanonicalizationState canonicalizationState;
```

Reply:
I'll add State to the name to distinguish it from some hypothetical stand-alone lifetime representation